### ISSUE TITLE
Markdown outline and folding support

### ISSUE THREAD

**Issue by StephenSmithwick** [zed-industries/zed#4924]
### Check for existing issues

- [X] Completed

### Describe the feature

Currently the html editor breaks editor content into a hierarchy which can be folded (hidden) by clicking on the carrot(`⌄`) in the margin.  

Can we have a similar capability added to markdown.  

The suggested behaviour would be similar to vscode's current solution:

```
⌄ # Header 1
  Some content
⌄ ## Sub Header
  Some more content
⌄ ## Header 2
  The final content
```
1. Clicking on carrot beside `Header 1` would hide all content below `Header 1` up to `Header 2`
2. Clicking on carrot beside `Sub Header` would hide content below `Sub Header` up to `Header 2`
3. Editing the header line can be edited in place without expanding. 
4. Editing the line below the hidden content would re-expand the hidden content

**Comment by notpeter** [comment]
More Markdown folding ideas:
- fenced code blocks <kbd>```</kbd>
- entire lists (currently only nested lists can be folded under their parent)
- tables
- definition lists

**Comment by aeryncaen** [comment]
Would love to see this, especially since I've started using Zed AI for documentation generation.

**Comment by opsnull** [comment]
I have lots of large markdown files, the outline and folding supporting is great to see。

**Comment by opsnull** [comment]
any planning for this？

**Comment by tv42** [comment]
See #22703 for likely implementation mechanism.

(I cannot emphasize enough how much I crave this feature! Many of my documents are just unwieldy to scroll back and forth in.)

**Comment by detjensrobert** [comment]
Not stale, latest tag 0.212 still doesn't have header folding.

**Comment by miguelraz** [comment]
Sorry for the noise, the `stalebot` upstream went haywire.

Please disregard it.

**Comment by aplatypus** [comment]
I want to check-in: is  this feature is still a work in progress?

The google AI bot seems to think markdown heading collapse is supported with:

- editor: fold
- editor: unfold

commands.  

I was unsuccessful fold/unfold some markdown file headings on version: 

- Zed 0.215.3 
  fda162bf3d9a11d18e07cbf2b488829450135d34
  0.215.3-90+fda162bf3d9a11d18e07cbf2b488829450135d34 

It is definitely something worthwhile in my book.

**Comment by Lukapetro** [comment]
would be nice to have this, easier navigating generated plans

**Comment by jeremychone** [comment]
@notpeter Yes, it’s based on indentation, which is why, in code, comments don’t get folded, even at level 0.

I really hope this isn’t too deep in the code design, and that we can have "syntax" folding someday.
